### DISCUSSION TITLE
Devnet Pre-Release 1.3.0alpha2

### DISCUSSION THREAD

**Discussion by lk86** [MinaProtocol/mina#10023] [Announcements]
The O(1) Labs team is proud to announce that we have additional features from the `compatible` branch ready for alpha testing on Devnet in the form of this 1.3.0alpha2 release. The scale of the changes since 1.2.2 is massive after months of work across many different parts of the application, and we have not yet fully documented those changes beyond what git will show. Look out for more thorough release notes in the next beta if not before. Below you can read the full scope of changes between 1.3.0alpha1 and 1.3.0alpha2.

Of the changes included in 1.3.0alpha2, we are most hopeful that the staged ledger diff optimizations in #9782 will help keep nodes in sync with the tip of the chain, even under higher transaction load.

If you run into any issues, please include `Release: 1.3.0alpha2` in the name of any issues you create. Please report any issues you encounter [here](https://github.com/MinaProtocol/mina/issues/new/choose) or join the [Discussion](https://github.com/MinaProtocol/mina/discussions/)!

### New Features for alpha2:
- Initial support for bitswap in libp2p helper #9626
- Fix SetGatingConfig RPC banning behavior #9895
- Choose the best VRF evaluation for a given slot rather than the first #8327
- Optimized Staged Ledger Diff Application #9782 #10018

### Testing Improvements:
- Catchup Failure Unit Tests #9889
- Utility File Integration Tests #9955
- Fix race condition in Integration Tests #10009
- Peer Graph Connectivity Integration Test #9977
- Remove verbose log #10007
- Unit test for SetGatingConfig #9895
- Unit test for sparse ledger subset extraction #9782

### Archive Node:
Some important archive node changes, including a new database schema, are not yet included in this release, so we advise anyone that is waiting for the 1.3.0+ archive node to wait for a future alpha with the final database schema and more complete migration instructions.

### Highlighted New Features introduced in alpha1:
- This version supports both MINA_ environment variables and the old CODA_ prefix, but the 1.3.X will be the last releases to support CODA_. We recommend you take this opportunity to use MINA_ everywhere going forward (example: MINA_PRIVKEY_PASS, MINA_CLIENT_TRUSTLIST, etc.)
- Support for Debian Buster in addition to Ubuntu18.04/ Debian Stretch. Official support for 20.04 is coming soon, hopefully before 1.3.0 stable. To use the new buster-based package or images, replace `stretch` with `buster` in either the docker image tag or the debian repository setup command.

### Complete Changelog since 1.3.0alpha1:
The change set is far too large for github to display, but the github comparison still shows the immense scale of 1.3.0 and why we are so excited to get it out into your hands to test.
https://github.com/MinaProtocol/mina/compare/1.3.0alpha1...1.3.0alpha2
For more information about the changes included in 1.3.0alpha1, see the [previous release notes](https://github.com/MinaProtocol/mina/releases/tag/1.3.0alpha1)

### Debian Packages:
**Debian Repository:**
First, set up and update the `alpha` Debian Repository for your platform (`stretch` or `buster`, ubuntu 20.04 coming soon)
```
echo "deb [trusted=yes] http://packages.o1test.net stretch alpha" | sudo tee /etc/apt/sources.list.d/mina-alpha.list
sudo apt-get update
```
Then, install the package(s) that you need:
**Mina Daemon:** `sudo apt-get install -y mina-devnet=1.3.0alpha2-2717385`
**Mina Archive:** `NOT READY YET - see the note above for more details`

### Docker Images:
**Mina Daemon:** `minaprotocol/mina-daemon:1.3.0alpha2-2717385-stretch-devnet`
or for a Debian Buster image: `minaprotocol/mina-daemon:1.3.0alpha2-2717385-buster-devnet`
**Mina Archive:** `NOT READY YET - see the note above for more details`

### Sandbox Node: For testing in an isolated, single-node network without snarks
As of 1.2.0, the demo functionality is built into the standard daemon container, via the environment variable `RUN_DEMO=true`.
Just `docker run -it --name mina-demo -e RUN_DEMO=true minaprotocol/mina-daemon:1.3.0alpha2-2717385-buster-devnet` and go!

### Step by Step Guide:
Check out [our documentation](https://docs.minaprotocol.com/en/advanced/connecting-devnet) for complete instructions on using this version to connect to Devnet. *Docker only* Make sure to run with `--peer-list-url https://storage.googleapis.com/seed-lists/devnet_seeds.txt`.

If you are running the correct version on the correct network, `mina client status` will show:
```
Chain id:     b6ee40d336f4cc3f33c1cc04dee7618eb8e556664c2b2d82ad4676b512a82418
Git SHA-1:    271738546ff38127772494949de5c23fb6336727
```

<hr /><em>This discussion was created from the release <a href='https://github.com/MinaProtocol/mina/releases/tag/1.3.0alpha2'>Devnet Pre-Release 1.3.0alpha2</a>.</em>

**Comment by scarletbright** [comment]
two of my 1.3.0alpha2 nodes are getting stuck on the same block height - 100320
restarting and removing .mina-config doesn't help! it's reaching 100320 and that's it again...
have never seen something like this before

**Comment by kakashi0x0** [comment]
I encountered the same issue what @scarletbright mentioned. Kept it online on mainnet for approx 18 hours. Couldn't sync after multiple attempts but produced few block on the fork. Noticed **new** activity in the logs like below. Logs exported and saved, will share if required.

```
Jan 17 23:19:30 mina[10966]: 2022-01-17 22:19:30 UTC [Faulty_peer] Banning peer $sender_id until "2022-01-18 22:19:30.929599Z" due to $action
Jan 17 23:19:30 mina[10966]:         sender_id: {
Jan 17 23:19:30 mina[10966]:   "host": "178.170.42.10",
Jan 17 23:19:30 mina[10966]:   "peer_id": "12D3KooWRoyruvBxFyNKWsXKBpjWiBwy4hEQ6NkEM2BoKyMpAMrx",
Jan 17 23:19:30 mina[10966]:   "libp2p_port": 8302
Jan 17 23:19:30 mina[10966]: }
Jan 17 23:19:30 mina[10966]:         action: "Trust_system.Actions.Gossiped_invalid_transition (invalid staged ledger diff: incorrect staged ledger hash && snarked ledger hash)"
```

**Comment by vaddec-everstake** [comment]
the last few days, the node cannot exit the "catchup" After a manual reboot, it works for several hours and returns back to the "catchup"

![image](https://user-images.githubusercontent.com/80746090/150510803-efdde0bf-c950-402d-b62a-211e358374ac.png)
